src/modules/Map/InteractiveMap/components/Legend/Legend.js

```diff
@@ -106,23 +110,21 @@ export const Legend = ({
             translate,
             label,
             boundaries,
-            index,
-            items.length,
             rounding,
           );
 
           return (
             <div
-              key={`${computedLabel}${color}`}
+              key={`${computedLabel}${color}${diameter}`}
               className={`tf-legend__item tf-legend__item--${shape} tf-legend__item--${position}`}
             >
               <div
                 className="tf-legend__symbol-container"
                 style={{ width: shape === 'circle' ? biggestDiameter : DEFAULT_DIAMETER * 2 }}
               >
-                {shape === 'circle'
-                  ? <Circle color={color} size={diameter} />
-                  : <Rect color={color} size={DEFAULT_DIAMETER} />}
+                {(shape === 'circle') && <Circle color={color} size={diameter} />}
+                {(shape === 'line') && <Line color={color} size={DEFAULT_DIAMETER} strokeWidth={strokeWidth} />}
+                {(shape === 'square') && <Rect color={color} size={DEFAULT_DIAMETER} />}
```

Comment:
Another way to write this could be:
*(I don't know if it's worth it or not. Probably not)*
```suggestion
                {{
                  circle: <Circle color={color} size={diameter} />,
                  line:  <Line color={color} size={DEFAULT_DIAMETER} strokeWidth={strokeWidth} />,
                  square:  <Rect color={color} size={DEFAULT_DIAMETER} />,
                }[shape]}
```

Reply:
The problem is the error if the configuration contains error and the shape is bad. With previous version there is no display with the new one we have an error.